plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/TimestampDecoder.java

```diff
@@ -97,11 +109,13 @@ else if (value instanceof Number) {
             implements DecoderDescriptor
     {
         private final String path;
+        private String patterns;
```

Comment:
How about changing `patterns` to `pattern` across the board wherever used? As we are setting/getting one pattern only.

Reply:
I have already made the modfication. However actually the String may refers to multiple patterns in elasticsearch mappings such as "strict_date_option_time||epoch_millis"

Reply:
Yes, I could see. You are using `String.join("||", formats)`.  Leave it `patterns` then. You can also revert others back to `patterns`